core/src/main/java/com/yahoo/imapnio/async/data/ExtensionListInfo.java

```diff
@@ -0,0 +1,93 @@
+package com.yahoo.imapnio.async.data;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+
+import com.sun.mail.iap.ParsingException;
+import com.sun.mail.imap.protocol.IMAPResponse;
+import com.sun.mail.imap.protocol.ListInfo;
+
+/**
+ * This class provides the result from LIST-EXTENDED command. It extends from ListInfo so clients can continue use the data from ListInfo. The public
+ * methods provided here allows callers to easily identify if server sends the specific interested attribute or not. The attributes supported are
+ * listed in RFC 5258.
+ */
+public class ExtensionListInfo extends ListInfo {
+    /**
+     * Extended attributes supported for LIST-EXTENDED capability, RFC 5258.
+     */
+    public enum ExtendedListAttribute {
+        /** HasChildren attribute. */
+        HAS_CHILDREN,
```

Comment:
can you make the strings part of the attribute HAS_CHILDREN("\\HasChildren")

Reply:
Done. Fixed!